src/modules/analytics/AnalyticsContext.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+import { Component, type Node } from 'react';
+import PropTypes from 'prop-types';
+
+import type { ObjectType } from './types';
+
+const ContextTypes = {
+  getAnalyticsContext: PropTypes.func,
+};
+
+type Props = {
+  data: ObjectType,
+  children: Node,
+};
+
+export default class AnalyticsContext extends Component<Props> {
+  static contextTypes = ContextTypes;
+
+  static childContextTypes = ContextTypes;
+
+  getChildContext = () => ({
+    getAnalyticsContext: this.getAnalyticsContext,
+  });
+
+  getAnalyticsContext = () => {
+    const { data } = this.props;
+    const { getAnalyticsContext } = this.context;
+    const ancestorData =
+      (typeof getAnalyticsContext === 'function' && getAnalyticsContext()) ||
+      [];
+    return [...ancestorData, data];
+  };
+
+  render() {
+    return this.props.children;
```

Comment:
Can these be wrapped with `React.Children.only` so we get better error reporting when there are multiple children?